postgres/postgres.go

```diff
@@ -67,6 +75,100 @@ func (ds Datastore) AddFilter(host string, newFilter *models.Filter) (models.Fil
 	return *newFilter, nil
 }
 
+// AddFilterDimension adds a dimension for a given dataset to be stored in postgres
+func (ds Datastore) AddFilterDimension(dimensionObject *models.AddDimension) error {
+	tx, err := ds.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	// Check filter exists
+	row := ds.getFilter.QueryRow(dimensionObject.FilterID)
```

Comment:
Should this be wrapped around the transaction? eg `tx.Stmt(ds.getFilte).QueryRow(dimensionObject.FilterID)`